woocommerce/wc-admin
# Migrate reviews panel to home screen

Fixes #5240 

The bulk of the code is migrating the reviews panel to the home screen. Currently there is a decent amount of overlapping logic between the reviews header panel and the home screen panel.
The home screen panel contains several actions (approve, set as spam, delete), these actions had to be added to the reviews data store - I added the updateReview and deleteReview actions.
Made use of `invalidateResolutionForStoreSelector` to re-trigger updates after update or deletion.

Updated the rating component to allow for custom icons to be passed in (as strings), this way I could show both the normal star, and the outline star as shown in the wireframe.

### Accessibility


### Screenshots

![review-panel](https://user-images.githubusercontent.com/2240960/100007652-bf28b500-2da2-11eb-9cbb-d628aea49bf3.gif)

### Detailed test instructions:

- Create new store and add several products, make sure you have the [products reviews](https://docs.woocommerce.com/document/configuring-woocommerce-settings/#section-9) enabled.
- Go to the public product page and add some comments (in incognito so the comments show up as 'pending')
- Go to the Woo home screen
- You should see a `Reviews to moderate` on the left, with the number of pending reviews (Reviews header panel should not be shown - awaiting feedback if just on home screen)
- You should be able to Approve, mark as spam, or delete a review, a popup will appear that allows you to 'undo' this action
- If no reviews are needing review it should not be allowed to collapse.

UPDATE:
There are a couple small styling changes that depend on Jeff's updates: https://github.com/woocommerce/woocommerce-admin/pull/5729/files#diff-beff577e21ba5ae2d55e05bdfcbcc94c7da812eb81f26b5da415b870d79d2c82
I believe the only thing that is noticeable is the margin between the icon and the subtitle.
